Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPITabsCocoa.mm

```diff
@@ -422,6 +422,28 @@ static inline String toMIMEType(WebExtensionTab::ImageFormat format)
     tab->toggleReaderMode(WTFMove(completionHandler));
 }
 
+void WebExtensionContext::tabsSendMessage(WebExtensionTabIdentifier tabIdentifier, String messageJSON, std::optional<WebExtensionFrameIdentifier> frameIdentifier, const WebExtensionMessageSenderParameters& senderParameters, CompletionHandler<void(std::optional<String> replyJSON, WebExtensionTab::Error)>&& completionHandler)
+{
+    auto tab = getTab(tabIdentifier);
+    if (!tab) {
+        completionHandler(nullString(), toErrorString(@"tabs.sendMessage()", nil, @"tab not found"));
+        return;
+    }
+
+    WebProcessProxySet processes = tab->processes(WebExtensionEventListenerType::RuntimeOnMessage);
+    if (processes.isEmpty()) {
+        completionHandler(std::nullopt, std::nullopt);
+        return;
+    }
+
+    ASSERT(processes.size() == 1);
+    auto process = processes.takeAny();
```

Comment:
does this just return any element in the set?

Reply:
Yep